### PR TITLE
Add a way to change the position of the cat.

### PR DESCRIPTION
**Pull request by slendidev**
This simple PR adds the ability to further customize the cat by allowing the user to change its position. This option can be found in the launcher's settings page.

### CODE REVIEW

File: launcher/ui/pages/global/LauncherPage.ui [hunk 1]
```diff
@@ -38,7 +38,7 @@
       <enum>QTabWidget::Rounded</enum>
      </property>
      <property name="currentIndex">
-      <number>0</number>
```
Comment (ghost) [inline launcher/ui/pages/global/LauncherPage.ui]: Change this back to 0. This is a weird thing with Qt Designer.

**Review by ghost — CHANGES_REQUESTED** [review]
Does this actually change the position? I don't see any code that does that

**Comment by slendidev** [comment]
> Does this actually change the position? I don't see any code that does that

Yes, look at the setStylesheet code.

**Comment by LennyMcLennington** [comment]
> > Does this actually change the position? I don't see any code that does that
> 
> Yes, look at the setStylesheet code.

There is no such code. You probably committed without using `git add` to stage whichever changes you're talking about.

**Comment by LennyMcLennington** [comment]
I don't think this will work properly with translations, and what if someone edits the config file and accidentally causes some invalid style sheet? There should be some form of validation, and make sure it isn't directly relying on the text of the combobox because that can be translated.

**Review by LennyMcLennington — CHANGES_REQUESTED** [review]
I don't think this will work properly with translations, and what if someone edits the config file and accidentally causes some invalid style sheet? There should be some form of validation, and make sure it isn't directly relying on the text of the combobox because that can be translated.

**Comment by ghost** [comment]
Stop force pushing, it messes up the commit history. Keep it clean and do one commit at a time, we can squash it later.

**Comment by slendidev** [comment]
OK.

**Comment by LennyMcLennington** [comment]
@xslendix force pushing in your own branch is fine in a lot of cases, I don't care to see a whole new commit for someone to change 1 character for a typo on a previous commit, and stuff like that... It doesn't "mess up" history, it rewrites history which shouldn't matter at all since it's your own personal branch not one that people are actively collaborating on, @binex-dsk is just being his regular self by saying silly nonsense.

**Review by LennyMcLennington — CHANGES_REQUESTED** [review]
If the text in the combo box is translatable then you shouldn't be using currentText to determine what is written into the settings... and it *should* be translatable. There are legitimate uses for the currentText method, this seems more like an abuse than a legitimate use.

If you want to store some internal data about the combobox item, you can do that too... The user-facing text is just not the right way to do it. E.g. `ui->comboBox->addItem(tr("Top Right"), "top right");` that way you have some internal data attached to the item which is the string "top right", but the user-facing text will say "Top Right" translated into whatever language they have selected. You can retrieve it with currentData or something like that.

**Comment by ghost** [comment]
> @xslendix force pushing in your own branch is fine in a lot of cases, I don't care to see a whole new commit for someone to change 1 character for a typo on a previous commit, and stuff like that... It doesn't "mess up" history, it rewrites history which shouldn't matter at all since it's your own personal branch not one that people are actively collaborating on, @binex-dsk is just being his regular self by saying silly nonsense.

It only makes sense if you change 1 character and nothing else, but if you make a decently large change then IMO it makes more sense to do a normal commit. I'd rather directly see what you changed, not have to fish for what did change. I just think it's better to keep commits consistent for the PR and then squash during the merge.

**Comment by slendidev** [comment]
Not gonna lie, I am going to agree with Lenny on this one. There is no point
on worrying about keeping the commit history clean when changes are on
different branch and in a PR that can change at any time. Not only this, but
you can also have a coherent look at the final commits, and prevents merging
accidentally before the commit history get figured out.

I will continue pushing with -f since it's not that big of a deal and I don't
think there's any real point in keeping the git history clean when you are
going to rebase at the end anyway.

**Comment by ghost** [comment]
> Not gonna lie, I am going to agree with Lenny on this one. There is no point on worrying about keeping the commit history clean when changes are on different branch and in a PR that can change at any time. Not only this, but you can also have a coherent look at the final commits, and prevents merging accidentally before the commit history get figured out.
> 
> I will continue pushing with -f since it's not that big of a deal and I don't think there's any real point in keeping the git history clean when you are going to rebase at the end anyway.

Again, I can't tell what you're changing when you keep force pushing.

**Comment by LennyMcLennington** [comment]
> Not gonna lie, I am going to agree with Lenny on this one. There is no point on worrying about keeping the commit history clean when changes are on different branch and in a PR that can change at any time. Not only this, but you can also have a coherent look at the final commits, and prevents merging accidentally before the commit history get figured out.
> 
> I will continue pushing with -f since it's not that big of a deal and I don't think there's any real point in keeping the git history clean when you are going to rebase at the end anyway.



> > @xslendix force pushing in your own branch is fine in a lot of cases, I don't care to see a whole new commit for someone to change 1 character for a typo on a previous commit, and stuff like that... It doesn't "mess up" history, it rewrites history which shouldn't matter at all since it's your own personal branch not one that people are actively collaborating on, @binex-dsk is just being his regular self by saying silly nonsense.
> 
> It only makes sense if you change 1 character and nothing else, but if you make a decently large change then IMO it makes more sense to do a normal commit. I'd rather directly see what you changed, not have to fish for what did change. I just think it's better to keep commits consistent for the PR and then squash during the merge.

I don't think squashing is good. I think that there should ideally be one commit for each "thing" that is fixed/changed/added. Squashing complicates going through the history granularly to find exact change I'm looking for from a large merge. If you add something and then you find a small oversight in that addition and fix that without changing the overall "thing" that the commit is supposed to do, I would amend/fixup that commit and force push.

There is an argument to be made for it to not be force pushed until it's no longer WIP since it's easier to see what's changed since the last review, but other than that I don't really see it. I would still expect him to clean up the history and get rid of all the redundant commits that were created when he responded to reviewers' requests for changes before the PR would finally be merged. I would prefer the commit history to be meaningful than to be:

1. implement some functionality X
2. oops I forgor something in feature X
3. Implement some functionality Y
4. I changed something in feature X cos a reviewer asked me to
5. shit I forgot to add this file to the cmakelists
6. changed something again cos another review
7. changed something again AGAIN cos another review

Sure that's fine when it's still WIP but I'd expect those to be force pushed out of existence to just 

1. Implement functionality x
2. Implement functionality y

or something like that before it actually gets merged unless there's a really good reason for one of the others to stay like if it's actually significantly different in design or scope from the other two

**Review by LennyMcLennington — APPROVED** [review]

**Review by ghost — APPROVED** [review]
I haven't gotten the chance to test but overall this looks good.

**Comment by JosTheDude** [comment]
The commit name LMAO

**Pull request merged**